### ISSUE TITLE
Upcoming breaking change regarding configGroupNesting

### ISSUE THREAD

**Issue by magnouvean** [nix-community/plasma-manager#93]
Edit: After #94 the format has changed again. The rest of comment has been updated to address this.

There are two new changes regarding the format of configurations for the `file` module. One is that all keys in `configFile`, `dataFile` and `file` require the `value` property, meaning, for example. what before was:
```nix
programs.plasma = {
  configFile = {
    "krunnerrc"."General"."FreeFloating" = true;
    "kwinrc"."Desktops"."Number" = 8;
    "kwinrc"."Desktops"."Rows" = 2;
  };
};
```
now must be:
```nix
programs.plasma = {
  configFile = {
    "krunnerrc"."General"."FreeFloating".value = true;
    "kwinrc"."Desktops"."Number".value = 8;
    "kwinrc"."Desktops"."Rows".value = 2;
  };
};
```
We also had to remove the recursive attribute config format (even though it was newly introduced), because it had some limitations with extensibility.

The other difference is that the new delimiter for nested groups is `/`, which was changed from `.`. This means that where you before had:
```nix
programs.plasma.configFile."group.subgroup1.subgroup2"."key" = "value";
```
you now will have to use:
```nix
programs.plasma.configFile."group/subgroup1/subgroup2"."key".value = "value";
```
This also means that if you don't have to escape periods anymore, they can just be left as is, but if you for some reason need `/` in your groups, you can escape this by `part1\\/part2\\/part3`, which will translate to a group named `part1/part2/part3` (not that there should be any reason to have `/` in your groups for kde plasma config-files anyway).

**Comment by magnouvean** [comment]
Thanks to #144 by @quentinmit you now can skip the `.value` part of the key if you don't want to tweak `immutability` or `persistence` or other attributes. In other words
```{nix}
programs.plasma.configFile.group.key = "value";
```
will be equivalent to
```{nix}
programs.plasma.configFile.group.key.value = "value";
```

I also said in another issue that I would step aside from the project due to switching distros, but that probably won't happen anyway as I did find out:
1. An OK solution to my software needs (for development)
2. Other distros are hardly good options for my needs as well

In other words I'll continue to be involved in the project in the future :)

I'll close this now since this isn't all that relevant anymore with the latest addition.
